pennylane/devices/null_qubit.py

```diff
@@ -277,10 +278,28 @@ def _max_workers(self):
         """No-op property to allow for borrowing DefaultQubit.preprocess without AttributeErrors"""
         return None
 
+    # pylint: disable=cell-var-from-loop
     def preprocess(
         self, execution_config=DefaultExecutionConfig
     ) -> Tuple[TransformProgram, ExecutionConfig]:
         program, _ = DefaultQubit.preprocess(self, execution_config)
+        for t in program:
+            if t.transform == decompose.transform:
+                original_stopping_condition = t.kwargs["stopping_condition"]
+
+                def new_stopping_condition(op):
+                    return (not op.has_decomposition) or original_stopping_condition(op)
+
+                t.kwargs["stopping_condition"] = new_stopping_condition
+
+                original_shots_stopping_condition = t.kwargs.get("stopping_condition_shots", None)
```

Comment:
Is this done because the stopping condition for when `Shots=None` is different?

Reply:
Yep! In some cases there's a different stopping condition for finite shots.